WS2012R2/lisa/setupscripts/CORE_reload_modules.ps1

```diff
@@ -106,8 +106,8 @@ function CheckResult()
             Start-Sleep -s 10
             $attempts--
             Write-Output "Info : Attempt number ${attempts}"
-            if ((Get-VMIntegrationService $vmName | ?{$_.name -eq "Heartbeat"}).PrimaryStatusDescription -eq "Lost Communication") {
-                Write-Output "Error : Lost Communication to VM" | Out-File -Append $summaryLog
+            if ((Get-VMIntegrationService $vmName -ComputerName $hvServer | ?{$_.name -eq "Heartbeat"}).PrimaryStatusDescription -match "No Contact|Lost Communication" ) {
+                Write-Output "Error : Lost Communication or No Contact to VM, maybe vm reboots" | Out-File -Append $summaryLog
```

Comment:
can you please also introduce here a stop-vm in force mode?
if the VM will get stuck, the lisa method of shutting down the VM might fail, as the VM is unresponsive and cannot process a graceful shutdown.

Verified it with:
stop-vm -vmName $vmName -computername $hvServer -force -turnoff

Reply:
Correct, thank you for your comment, will update